src/execution_plan/optimizations/seek_by_id.c

```diff
@@ -128,15 +128,23 @@ static void _UseIdOptimization(ExecutionPlan *plan, OpBase *scan_op) {
 			}
 
 			_setupIdRange(rel, id, reverse, &minId, &maxId, &inclusiveMin, &inclusiveMax);
-			opNodeByIdSeek = NewNodeByIdSeekOp(scan_op->plan, node, minId, maxId,
-											   inclusiveMin, inclusiveMax);
-
-			// Managed to reduce!
-			ExecutionPlan_ReplaceOp(plan, scan_op, opNodeByIdSeek);
-			ExecutionPlan_RemoveOp(plan, (OpBase *)filter);
-
+			/* Don't replace label scan, but set it to have range query.
+			 * Issue 818 https://github.com/RedisGraph/RedisGraph/issues/818
+			 * This optimization caused a range query over the entire range of ids in the graph
+			 * regardless to the label. */
+			if(scan_op->type == OPType_NODE_BY_LABEL_SCAN) {
+				NodeByLabelScan *label_scan = (NodeByLabelScan *) scan_op;
+				NodeByLabelScanOp_IDRange(label_scan, minId, inclusiveMin, maxId, inclusiveMax);
+			} else {
+				OpBase *opNodeByIdSeek = NewNodeByIdSeekOp(scan_op->plan, node, minId, maxId,
+														   inclusiveMin, inclusiveMax);
+
+				// Managed to reduce!
+				ExecutionPlan_ReplaceOp(plan, scan_op, opNodeByIdSeek);
+				OpBase_Free(scan_op);
+			}
 			// Free replaced operations.
-			OpBase_Free(scan_op);
+			ExecutionPlan_RemoveOp(plan, (OpBase *)filter);
 			OpBase_Free((OpBase *)filter);
 
 			break; // Exit loop.
```

Comment:
This is true on master as well, but I think this entire optimization isn't working as it's supposed to, in that it will only replace 1 filter rather than all possible filters:
```
$ ge G "MATCH (a) WHERE ID(a) >= 2 AND ID(a) < 8 RETURN a.val"
1) "Results"
2) "    Project"
3) "        Filter"
4) "            NodeByIdSeek | (a)"
```
It should replace all possible filters like the index optimization does: https://github.com/RedisGraph/RedisGraph/blob/master/src/util/range/numeric_range.c#L56-L102 .

Since this is an issue on master, we can address it outside of this PR, but should do so soon.